compiler/noirc_evaluator/src/ssa/ssa_gen/program.rs

```diff
@@ -17,6 +17,10 @@ pub(crate) struct Ssa {
     /// This mapping is necessary to use the correct function pointer for an ACIR call,
     /// as the final program artifact will be a list of only entry point functions.
     pub(crate) entry_point_to_generated_index: BTreeMap<FunctionId, u32>,
+    /// Flag that determines when to inline functions marked to not use predicates.
+    /// This is currently defaulted to `false` and should be set to true by after completion
+    /// of the flattening pass which sets side effects.
+    pub(crate) past_flattening_pass: bool,
```

Comment:
```suggestion
```
I think it is better to have an alternate entry point into the inlining pass (e.g. `inline_no_predicates` or similar) than to keep track of an extra bit of internal state which is only relevant for enabling one pass.

Reply:
I switched it over